catalog/video/video-mpegts-stream/src/main/java/org/codice/alliance/video/stream/mpegts/plugins/UpdateParent.java

```diff
@@ -0,0 +1,153 @@
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.alliance.video.stream.mpegts.plugins;
+
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+import org.codice.alliance.video.stream.mpegts.Context;
+import org.codice.alliance.video.stream.mpegts.framework.CatalogUpdateRetry;
+import org.codice.alliance.video.stream.mpegts.rollover.RolloverActionException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import ddf.catalog.data.Metacard;
+import ddf.catalog.operation.UpdateRequest;
+import ddf.catalog.operation.impl.UpdateRequestImpl;
+
+/**
+ * Update the parent metacard by calling an {@link UpdateField} and then by calling the
+ * catalog framework.
+ */
+public class UpdateParent implements FindChildrenStreamEndPlugin.Handler {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(UpdateParent.class);
+
+    private static final long INITIAL_RETY_MILLISECONDS = TimeUnit.SECONDS.toMillis(1);
+
+    private static final long MAX_RETRY_MILLISECONDS = TimeUnit.MINUTES.toMillis(5);
+
+    private final UpdateField updateField;
+
+    private CatalogUpdateRetry catalogUpdateRetry = new CatalogUpdateRetry();
+
+    public UpdateParent(UpdateField updateField) {
+        this.updateField = updateField;
+    }
+
+    public void setCatalogUpdateRetry(CatalogUpdateRetry catalogUpdateRetry) {
+        this.catalogUpdateRetry = catalogUpdateRetry;
+    }
+
+    @Override
+    public void handle(Context context, Metacard parent, List<Metacard> children) {
+        updateField.updateField(parent, children, context);
+    }
+
+    @Override
+    public void end(Context context, Metacard parentMetacard) {
+
+        updateField.end(parentMetacard, context);
+
+        UpdateRequest updateRequest = createUpdateRequest(parentMetacard);
+
+        update(context, parentMetacard, updateRequest);
+
+    }
+
+    private void update(Context context, Metacard parentMetacard, UpdateRequest updateRequest) {
+        try {
+            catalogUpdateRetry.submitUpdateRequestWithRetry(context.getUdpStreamProcessor()
+                            .getCatalogFramework(),
+                    updateRequest,
+                    context.getUdpStreamProcessor()
+                            .getMetacardUpdateInitialDelay(),
+                    INITIAL_RETY_MILLISECONDS,
+                    MAX_RETRY_MILLISECONDS,
+                    update -> {
+                        LOGGER.debug("updated parent metacard: newMetacard={}",
+                                update.getNewMetacard()
+                                        .getId());
+                        context.setParentMetacard(update.getNewMetacard());
+                    });
+        } catch (RolloverActionException e) {
+            LOGGER.debug("unable to update parent metacard: metacardId={}",
```

Comment:
Derived metacards don't get updated. But maybe there could be a larger discussion with @jlcsmith on what should happen if the parent doesn't get updated.